source/common/quic/envoy_quic_utils.cc

```diff
@@ -215,6 +217,25 @@ createConnectionSocket(const Network::Address::InstanceConstSharedPtr& peer_addr
   if (prefer_gro && Api::OsSysCallsSingleton::get().supportsUdpGro()) {
     connection_socket->addOptions(Network::SocketOptionFactory::buildUdpGroOptions());
   }
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.udp_set_do_not_fragment")) {
+    int v6_only = 0;
+    if (connection_socket->ipVersion().has_value() &&
+        connection_socket->ipVersion().value() == Network::Address::IpVersion::v6) {
```

Comment:
In the case of the listener socket up above, it looks like we don't call getsocketopt(). What's the difference which requires us to do so here?

Reply:
Whether a listen socket is v6 only or not purely depends on the configured listening address. If the address is v6 only, Envoy will call setsockopt() with V6ONLY option. But client connection sockets in addition to the configured address, also depends on platform (Android and iOS don't set the socket option V6ONLY even if the address is configured v6 only), so checking the option value here is more straight forward for client sockets.